rxandroidble/src/main/java/com/polidea/rxandroidble/internal/util/RxBleServicesLogger.java

```diff
@@ -0,0 +1,154 @@
+package com.polidea.rxandroidble.internal.util;
+
+import android.bluetooth.BluetoothDevice;
+import android.bluetooth.BluetoothGattCharacteristic;
+import android.bluetooth.BluetoothGattDescriptor;
+import android.bluetooth.BluetoothGattService;
+
+import com.polidea.rxandroidble.RxBleDeviceServices;
+import com.polidea.rxandroidble.internal.RxBleLog;
+import com.polidea.rxandroidble.utils.StandardUUIDsParser;
+
+import bleshadow.javax.inject.Inject;
+
+/**
+ * --------------- ====== Printing peripheral content ====== ---------------
+ * PERIPHERAL ADDRESS: AA-BB-CC-DD-EE-FF
+ * PERIPHERAL NAME: Device name
+ * -------------------------------------------------------------------------
+ * Primary Service - Weight Scale (ce029566-f9eb-11e7-8c3f-9a214cf093ae)
+ * Instance ID: 1111
+ * <p>
+ * -> Characteristics:
+ * * Weight Measurement (ce029962-f9eb-11e7-8c3f-9a214cf093ae)
+ * Properties: READ, WRITE, NOTIFY, INDICATE, etc...
+ * -> Descriptors
+ * * Client Characteristic Configuration (ce029bc4-f9eb-11e7-8c3f-9a214cf093ae)
+ * * Characteristic User Description (ce02a344-f9eb-11e7-8c3f-9a214cf093ae)
+ */
+public class RxBleServicesLogger {
+
+    private final CharacteristicPropertiesParser characteristicPropertiesParser;
+
+    @Inject
+    RxBleServicesLogger(CharacteristicPropertiesParser characteristicPropertiesParser) {
+        this.characteristicPropertiesParser = characteristicPropertiesParser;
+    }
+
+    public void log(RxBleDeviceServices rxBleDeviceServices, BluetoothDevice device) {
+        if (RxBleLog.isAtLeast(RxBleLog.VERBOSE)) {
+            RxBleLog.v("Preparing services description");
+            RxBleLog.d(prepareServicesDescription(rxBleDeviceServices, device));
```

Comment:
shouldn't it be `.v(...)` as well?

Reply:
I'm okay with `.d` level. I'd go for an additional flag if this would require an operation to be performed for logging, so the user could decide about the flow. In this case (passive) I'd stick with debug level. Would that be ok?

Reply:
Yup.